restord/ImageSessionServer.m

```diff
@@ -260,39 +260,50 @@ - (NSString *)apfsOSDisk {
 - (int)applyImage {
   NSString *path = self.image.localURL.path;
 
-  self.asr = [[NSTask alloc] init];
-  self.asr.launchPath = @"/usr/sbin/asr";
-  self.asr.arguments = @[ @"restore",
-                          @"--buffersize",
-                          @"16m",
-                          @"--source",
-                          path,
-                          @"--target",
-                          self.destination.path,
-                          @"--erase",
-                          @"--noprompt",
-                          @"--noverify",
-                          @"--puppetstrings" ];
-
-  // Set task environment.
-  NSMutableDictionary *environment = [[[NSProcessInfo processInfo] environment] mutableCopy];
-  environment[@"NSUnbufferedIO"] = @"YES";
-  self.asr.environment = environment;
-
-  // Create output pipe & file handle.
-  self.asr.standardError = self.asr.standardOutput = [[NSPipe alloc] init];
-  NSFileHandle *outputFh = [self.asr.standardOutput fileHandleForReading];
-  outputFh.readabilityHandler = ^(NSFileHandle *h) {
-    NSData *availableData = [h availableData];
-    [self processOutput:(NSData *)availableData];
-  };
-
-  // Launch and wait for exit.
-  [self.asr launch];
-  [self.asr waitUntilExit];
-
-  // Clear readability handler or the file handle is never released.
-  outputFh.readabilityHandler = nil;
+  int retries = (int)[[[NSUserDefaults alloc] initWithSuiteName:kPreferenceDomain] integerForKey:kASRAttempts];
+  if (retries <= 0) retries = 1;
+  NSLog(@"Will attempt ASR up to %d times.", retries);
+  for(int i = 0; i < retries; i++) {
```

Comment:
nit: Missing space after for
nit: Pre-increment i 
for (int i = 0; i < retries; ++i) {

Reply:
Done.  I don't think the compiler cares about pre-increment vs post.

Reply:
I think compilers care very much about them. I think the point you are trying to make is that, in the end the compiler optimizes away the additional instructions from i++. If so, I agree but choose to be explicit.